hadoop-project/pom.xml

```diff
@@ -1715,9 +1715,56 @@
           <artifactId>frontend-maven-plugin</artifactId>
           <version>${frontend-maven-plugin.version}</version>
         </plugin>
+        <plugin>
+          <groupId>org.xolstice.maven.plugins</groupId>
+          <artifactId>protobuf-maven-plugin</artifactId>
+          <version>${protobuf-maven-plugin.version}</version>
+          <extensions>true</extensions>
+          <configuration>
+            <protocArtifact>
+              com.google.protobuf:protoc:${protobuf.version}:exe:${os.detected.classifier}
+            </protocArtifact>
+            <attachProtoSources>false</attachProtoSources>
+          </configuration>
+          <executions>
+            <execution>
+              <id>src-compile-protoc</id>
+              <phase>generate-sources</phase>
+              <goals>
+                <goal>compile</goal>
+              </goals>
+              <configuration>
+                <includeDependenciesInDescriptorSet>false</includeDependenciesInDescriptorSet>
```

Comment:
in #1494 this execution is made to skip by default.
Reason is, every module can specifically enable this execution whenever required. Anyway every module needs to provide specific dependency information.

Reply:
I think for most sub modules, if it declares this plugin, it means we want to execute the compile goal, otherwise just do not declare the plugin. That's why I mean in the parent pom, we can declare skip=false for the compile execution. Anyway, can follow your solution. Not a big problem.